integration_test/cases/client_test.exs

```diff
@@ -127,26 +152,30 @@ defmodule ClientTest do
     try do
       P.run(pool, fn(conn) ->
         assert_receive :reconnected
-        spawn_link(fn() ->
-          assert P.run(pool, fn(_) -> :result end) == :result
-          send(parent, :done)
-        end)
         P.execute(conn, %Q{}, [:param])
-      end, [timeout: 0])
+      end, [timeout: 100])
     catch
       :throw, :oops ->
         :ok
     end
 
-    assert_receive :done
+    spawn_link(fn ->
```

Comment:
Don't need to spawn link here? Though comment above should handle the test and we dont need this.